android/src/main/java/com/swmansion/reanimated/NodesManager.java

```diff
@@ -234,25 +234,29 @@ public void runGuarded() {
   }
 
   private void onAnimationFrame(long frameTimeNanos) {
-    currentFrameTimeMs = frameTimeNanos / 1000000.;
+    double currentFrameTimeMs = frameTimeNanos / 1000000.;
 
-    while (!mEventQueue.isEmpty()) {
-      CopiedEvent copiedEvent = mEventQueue.poll();
-      handleEvent(copiedEvent.getTargetTag(), copiedEvent.getEventName(), copiedEvent.getPayload());
-    }
+    if (currentFrameTimeMs != lastFrameTimeMs) {
```

Comment:
It'd be more intuitive for me if we had `currentFrameTimeMs > lastFrameTimeMs` rather than just inequality check. We want to only do things if we are processing a new frame (that's later than the last processed frame). Would be also great to add a short comment about this check to make it cleaner.

Reply:
Resolved in: https://github.com/software-mansion/react-native-reanimated/pull/4579/commits/e67d9e1f2de8ad4a5acbcbd7da782911649f9d50